timesketch/lib/analyzers/contrib/misp_analyzer_test.py

```diff
@@ -0,0 +1,86 @@
+"""Test for MISP"""
+
+import copy
+import mock
+
+from flask import current_app
+
+from timesketch.lib.analyzers.contrib import misp_analyzer
+from timesketch.lib.testlib import BaseTest
+from timesketch.lib.testlib import MockDataStore
+
+SHA256_VALUE = "ac7233de5daa4ab262e2e751028f56a7e9d5b9e724624c1d55e8b070d8c3cd09"
+MISP_ATTR = {
+    "response": {
+        "Attribute": [
+            {
+                "event_id": "5",
+                "category": "Payload delivery",
+                "type": "filename",
+                "value": "test.txt",
+                "Event": {
+                    "org_id": "1",
+                    "id": "5",
+                    "info": "Hash Test",
+                    "uuid": "1f456b69-00c6-4fb0-8d92-709a0061b7d4",
+                },
+            },
+        ]
+    }
+}
+MATCHING_MISP = {"filename": "test.txt"}
+
+
+class TestMisp(BaseTest):
+    """Tests the functionality of the analyzer."""
+
+    def setUp(self):
+        super().setUp()
+        current_app.config["MISP_URL"] = "blah"
+        current_app.config["MISP_API_KEY"] = "blah"
+
+    @mock.patch("timesketch.lib.analyzers.interface.OpenSearchDataStore", MockDataStore)
+    @mock.patch(
+        "timesketch.lib.analyzers.contrib.misp_analyzer."
+        "MispAnalyzer.get_misp_attributes"
+    )
```

Comment:
Similar comment to the other test. This is pretty good but you really want your mock as far from your codebase as possible, so your tests cover as much code as they can (so mocking `requests.get` and playing with `mock_requests_get.return_value.json.return_value` is your best way to go.

Reply:
Fixed in 2927ba1189072618de1d0de9d0a4b59d257b1d78.